api/type.rb

```diff
@@ -216,6 +222,22 @@ def at_least_one_of_list
       @at_least_one_of
     end
 
+    # Checks that all properties that needs exactly one of their fields actually exist.
+    # This currently just returns if empty, because we don't want to do the check, since
+    # this list will have a full path for nested attributes.
+    def check_exactly_one_of
+      check :exactly_one_of, type: ::Array, default: [], item_type: ::String
+
+      return if @exactly_one_of.empty?
+    end
+
+    # Returns list of properties that needs at least one of their fields set.
+    def exactly_one_of_list
+      return [] unless @__resource
```

Comment:
Will `@__resource` ever not be true?

Reply:
I'm not all that familiar with Ruby, but printed out `@__resource`.  Looks like the entire resource definition from yaml, so probably not.  It's copied code, and I'm not sure it would be empty in any of the other places either, but maybe good practice to keep in place?  Due to my lack of Ruby knowledge, though, I have no strong opinions.

Reply:
Keeping it the same for consistency is good with me!

It's pretty unlikely to be empty- `__resource` is an instance variable that's always filled in by the resource's constructor. The `__` doesn't indicate anything special, it's just part of the var name.